designs/2020-rule-tester-debugger/README.md

```diff
@@ -0,0 +1,78 @@
+- Repo: eslint/eslint
+- Start Date: 2020-09-21
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: [CoryDanielson](https://github.com/corydanielson)
+
+# `RuleTester` debugger improvements
+
+## Summary
+
+Add a `before` hook to the valid/invalid code configuration, so that developers can inject a debugger into RuleTester before code is passed into the rule being tested.
```

Comment:
I couldn't immediately think of another use case that wasn't already solved by another API. I thought `before` and `after` would allow for performance timing, but ESLint has documentation about performance analysis on rules.